flink-core/src/main/java/org/apache/flink/api/connector/source/SplitEnumerator.java

```diff
@@ -29,7 +30,8 @@
  * 2. assign the splits to the source reader.
  */
 @PublicEvolving
-public interface SplitEnumerator<SplitT extends SourceSplit, CheckpointT> extends AutoCloseable {
+public interface SplitEnumerator<SplitT extends SourceSplit, CheckpointT>
```

Comment:
Similar as above, could add this default method override for `notifyCheckpointComplete()`.